finesse/gui/temp_control.py

```diff
@@ -41,138 +34,147 @@ def _create_controls(self) -> QGridLayout:
             QGridLayout: The layout containing the figure.
         """
         layout = QGridLayout()
+        self._btns = {"hot": QPushButton("Hot BB"), "cold": QPushButton("Cold BB")}
+        self._btns["hot"].clicked.connect(  # type: ignore
+            partial(self._btn_pressed, name="hot")
+        )
+        self._btns["cold"].clicked.connect(  # type: ignore
+            partial(self._btn_pressed, name="cold")
+        )
         self._create_figure()
 
-        layout.addWidget(self._canvas, 0, 0)
+        layout.addWidget(self._btns["hot"], 0, 0)
+        layout.addWidget(self._btns["cold"], 1, 0)
+        layout.addWidget(self._canvas, 0, 1, 2, 1)
 
         return layout
 
     def _create_figure(self) -> None:
-        """Creates the matplotlib figure to be contained within the panel.
-
-        Returns:
-            None
-        """
-        self._figure, self._ax1 = plt.subplots()
+        """Creates the matplotlib figure to be contained within the panel."""
+        self._figure, ax = plt.subplots()
+        self._ax = {"hot": ax}
         self._canvas = FigureCanvasQTAgg(self._figure)
 
-        t = [1040, 1050, 1060, 1070, 1080, 1090, 1100]
-        hot_bb_temp = [55, 57.5, 60, 62.5, 65, 67.5, 70]
-        cold_bb_temp = [1, 1.1, 1.15, 1.2, 1.3, 1.4, 1.5]
+        # Placeholder callback to test updating of figure
+        self._canvas.mpl_connect("button_press_event", self._update_figure)
+
+        self._figure_num_pts = 10
+        t = [None] * self._figure_num_pts
+        hot_bb_temp = [None] * self._figure_num_pts
+        cold_bb_temp = [None] * self._figure_num_pts
+
+        colours = plt.rcParams["axes.prop_cycle"]
+        hot_colour = colours.by_key()["color"][0]
+        cold_colour = colours.by_key()["color"][1]
 
-        self._bb_hot_line = self._ax1.plot(t, hot_bb_temp, color=[0, 1, 0])
-        self._ax1.set_xlabel("")
-        self._ax1.set_ylabel("HOT BB", color=[0, 1, 0])
-        self._ax1.set_xlim([1045, 1101.4])
-        self._ax1.set_ylim([20, 80])
+        self._ax["hot"].plot(t, hot_bb_temp, color=hot_colour)
+        self._ax["hot"].set_ylabel("HOT BB", color=hot_colour)
+        self._ax["hot"].set_xlim([1045, 1101.4])
+        self._ax["hot"].set_ylim([20, 80])
 
-        self._ax2 = self._ax1.twinx()
-        self._bb_cold_line = self._ax2.plot(t, cold_bb_temp, color=[1, 1, 0])
-        self._ax2.set_ylabel("COLD BB", color=[1, 1, 0])
-        self._ax2.set_ylim([0, 10])
+        self._ax["cold"] = self._ax["hot"].twinx()
+        self._ax["cold"].plot(t, cold_bb_temp, color=cold_colour)
+        self._ax["cold"].set_ylabel("COLD BB", color=cold_colour)
+        self._ax["cold"].set_ylim([0, 10])
 
         self._canvas.draw()
 
-    def _update_figure(self) -> None:  # , x, y) -> None:
-        """Updates the matplotlib figure to be contained within the panel.
+    def _btn_pressed(self, name) -> None:
```

Comment:
This needs a type annotation:

```suggestion
    def _btn_pressed(self, name: str) -> None:
```

Not sure why `mypy` didn't flag it!

Reply:
It's also good practice to name functions according to what they *do*, so in this case it could be `_toggle_axis_visibility` or something. (I'm actually not particularly consistent with this myself tbh :laughing:)